department-of-veterans-affairs/caseflow
# Sean C/APPEALS-17436

Resolves [APPEALS-17436](https://vajira.max.gov/browse/APPEALS-17436)

### Description
- Updated the "Return to VHA CAMO" task action and modal
- Updated the "Return to Program Office" task action and modal

### Acceptance Criteria
- [X] VHA Program Office user has the task action  "Return to VHA CAMO" available in their actions dropdown
- When selected a modal appears containing:
  - [X] Header: Return to VHA CAMO team
  - [X] Information text: Why are you returning? Provide any important context *THIS MAY CHANGE*
  - [X] Text field below information text
  - [X] Cancel button, and Return button
  - [X] The user is required to populate the text field before the "Return" button is enabled
- When the action is executed the following occurs:
  - [X] The appeal is removed from the Program Offices queue
  - [X] The appeal appears in the "Assigned" tab of the CAMO queue
  - [X] The user is redirected to the "Assigned" tab of the Program Offices queue
- [X] VISN user has the task action  "Return to Program Office" available in their actions dropdown
- When selected a modal appears containing:
  - [X] Header: Return to Program Office
  - [X] Information text: Why are you returning? Provide any important context *THIS MAY CHANGE*
  - [X] Text field below information text
  - [X] The user is required to populate the text field before the "Return" button is enabled
  - [X] Cancel button, and Return button
- When the action is executed the following occurs:
  - [X] The appeal is removed from the VISN queue
  - [X] The appeal appears in the "Assigned" tab of the Program Office queue
  - [X] The user is redirected to the "Assigned" tab of the VISN queue

### Before Testing
- Run these commands to make sure you have the appropriately setup appeals and take note of the veteran of each appeal:
```
	po = VhaProgramOffice.find(41)
	task = FactoryBot.create(:assess_documentation_task, assigned_to: po)
	task.appeal.veteran
	ro = VhaRegionalOffice.find(46)
	search_task = FactoryBot.create(:vha_document_search_task)
	assess_task = FactoryBot.create(:assess_documentation_task, parent: search_task, assigned_to: po)
	child_assess_task = FactoryBot.create(:assess_documentation_task, parent: assess_task, assigned_to: ro)
	child_assess_task.appeal.veteran
```

### Testing Plan
1. Log in as VHAPOUSER and go to "Your Queue"
2. Select "Community Care - Payment Operations Management cases", from the "Switch Views" dropdown select
3. Click on the last case of the queue, it should be the first appeal that you created before starting testing
4. Take note of the Veteran's name and ID
5. Select "Return to VHA CAMO", from the "Actions" dropdown
6. Enter anything in to the text field
7. Click the "Return" button
8. Switch users, log in as CAMOUSER and go to "Your Queue"
9. From the "Switch Views" dropdown select "VHA CAMO cases
10. Click on the Veteran's name and ID that you noted earlier
11. In the "Case Timeline" section click on the "View task instructions" and you should see that the task was canceled and the text you entered in the text field
12. Switch users, log in as VISNUSER and go to "Your Queue"
13. Select "VA New England Healthcare System cases", from the "Switch Views" dropdown select
14.  Click on the last case of the queue, it should be the second appeal that you created before starting testing
15. Select "Return to Program Office", from the "Actions" dropdown
16. Enter anything in to the text field
17. Click the "Return" button
18. Switch users, log in as VHAPOUSER and go to "Your Queue"
19. Select "Community Care - Payment Operations Management cases", from the "Switch Views" dropdown select
20. Click on the Veteran's name and ID that you noted earlier
21. In the "Case Timeline" section click on the "View task instructions" and you should see that the task was canceled and the text you entered in the text field


### User Facing Changes
 - [ X ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
<img width="499" alt="17436_Before_1" src="https://user-images.githubusercontent.com/110493538/235223677-eec56ed2-f18c-450e-8085-b5555959ce61.png">|<img width="499" alt="17436_After_1" src="https://user-images.githubusercontent.com/110493538/235223698-9478dbb5-2b3e-470d-be8e-313755488319.png">
<img width="497" alt="17436_Before_2" src="https://user-images.githubusercontent.com/110493538/235223728-9e93553e-139b-4a62-bbac-68472d24f7aa.png">|<img width="494" alt="17436_After_2" src="https://user-images.githubusercontent.com/110493538/235223749-d84f43ec-7cf7-446b-8d93-2f7c1ea578e9.png">


### Storybook Story
*For Frontend (Presentationa) Components*
